pkg/collector/corechecks/oracle-dbm/init.go

```diff
@@ -45,16 +47,23 @@ func (c *Check) init() error {
 	}
 
 	var i vInstance
-	// host_name is null on Oracle Autonomous Database
-	err = getWrapper(c, &i, "SELECT /* DD */ nvl(host_name, instance_name) host_name, version_full FROM v$instance")
+	err = getWrapper(c, &i, "SELECT /* DD */ host_name, instance_name, version_full FROM v$instance")
 	if err != nil {
 		return fmt.Errorf("%s failed to query v$instance: %w", c.logPrompt, err)
 	}
 	c.dbVersion = i.VersionFull
 	if c.config.ReportedHostname != "" {
 		c.dbHostname = c.config.ReportedHostname
 	} else {
-		c.dbHostname = i.HostName
+		if i.HostName.Valid {
+			c.dbHostname = i.HostName.String
+		} else {
+			// host_name is null on Oracle Autonomous Database
+			c.dbHostname = i.InstanceName
+		}
+	}
+	if i.HostName.Valid {
+		tags = append(tags, fmt.Sprintf("real_hostname:%s", i.HostName.String))
```

Comment:
so this would be an instance tag, and would appear on all events and metrics?

Reply:
Yes